### ISSUE TITLE
If a machine is decided returns the name/link of its decider

### ISSUE THREAD

**Issue by tcosmo** [bbchallenge/bbchallenge-api#2]
This is not completely trivial as it implies looking for the ID in the various index files containing the IDs of the machines decided by each machine (and they can be big, like around 1Go).

Fortunately, they are sorted so a dichotomy on the file cursor with `seek` should be efficient. But a bit of work and requires some testing.

**Comment by sligocki** [comment]
This feels like we are in need of some sort of database which could store various metadata for each TM and provide quick lookup. Then each time you run a decider, you can update the DB.

**Comment by tcosmo** [comment]
I went that way and I found no database system able to easily handle our 88M machines with metadata. In particular I tried PostgreSQL on my VPS setup and it would take 24 hours (not joking) just to load each machine with no metadata. Making request was awfully slow and not workable for the website.

This is why I opted for the simplest kind of databases: binary files. 

The seed database file (https://bbchallenge.org/method#download) defines an index over undecided machines and then, applying a decider only consists in giving the list of the IDs that it decided.

I maitain the negative set (i.e. set of machines that are not decided yet) on github as it is smaller: https://github.com/bbchallenge/bbchallenge-undecided-index

It is directly from this file that the website is able to give you the decided/undecided status efficiently.

This also allows to easily rollback when mistakes are made (which already happened).

People can also contribute to decide machines by PR with an updated index.
